tests/dataframe/test_iterrows_itertuples_pytest.py

```diff
@@ -0,0 +1,45 @@
+#  Licensed to Elasticsearch B.V. under one or more contributor
+#  license agreements. See the NOTICE file distributed with
+#  this work for additional information regarding copyright
+#  ownership. Elasticsearch B.V. licenses this file to you under
+#  the Apache License, Version 2.0 (the "License"); you may
+#  not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+# File called _pytest for PyCharm compatability
+
+from pandas.testing import assert_index_equal, assert_series_equal
+
+from tests.common import TestData
+
+
+class TestDataFrameIterrowsItertuples(TestData):
+    def test_iterrows(self):
+        ed_flights_iterrows = self.ed_flights().iterrows()
```

Comment:
```python
>>> x = ed_flights.iterrows()
>>> x
<generator object Operations.iterrows at 0x00000249AACC79E0>
>>> y = pd_flights.iterrows()
>>> y
<generator object DataFrame.iterrows at 0x00000249AACC7F90>
```

Thinking its not best to expose this as `Operations` .

Reply:
I think this can be achieved by returning df from internal methods and yielding from the 
`dataframe.iterrows()` and `dataframe.itertuples()` in `dataframe.py`